### DISCUSSION TITLE
How to use custom trained model in spacy3

### DISCUSSION THREAD

**Discussion by faisal-hearth** [explosion/spaCy#7396] [Help: Coding & Implementations] [answered]
I have trained a custom NER model and packaged it..
But I don't know how to import and use it.
I was using this [project](https://github.com/explosion/projects/tree/v3/pipelines/ner_demo) for help, it's nowhere mentioned how to use that trained model in actual python code.

**Comment by polm** [comment, marked as answer]
That's a good point! For that particular project here are two ways you can do it:

First, you can always load a model by specifying the directory where it's saved. In that project it's saved to `training/model-best`, so you can load the model like this:

    nlp = spacy.load('training/model-best')

Second, if you use the packaging command, you can distribute the model as a pip package. If you run `spacy project run package` you'll see a directory with the name `packages/en_ner_demo-0.0.0`. In this case the `en_ner_demo` is the package name, so you can install it like this:

    pip install -e packages/en_ner_demo-0.0.0

And after that you can load it in spaCy using `spacy.load("en_ner_demo")`, just like the official spaCy models. If you want to distribute the model you can package it as a single file using `python setup.py sdist` or other commands, just like a normal Python package.

**Reply by faisal-hearth** [reply]
Awesome it worked,
Thanks a lot @polm .
